### PR TITLE
Add pypartmc recipe from pypi to conda

### PR DESCRIPTION
**Pull request by a-v**
Using grayskull + pre-established pypi package for PyPartMC: https://pypi.org/project/PyPartMC

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/pypartmc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+requirements:
+  build:
+    - make
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}
+    - cmake
+    - flang  # [win]
+    - vs2019_win-64  # [win64]
+    - vs2019_win-32  # [win32]
```
Comment (isuruf) [inline recipes/pypartmc/meta.yaml:32]: Can you remove these?

File: recipes/pypartmc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  sha256: 8fb2d09b01724259e28f7dbbb6e48224852736a451234ef4133b394c405bce27
+
+build:
+  skip: true  # [win or py<37]
+  script: {{ PYTHON }} -m pip install . -vv
+  script_env:
+    - CMAKE_GENERATOR=Ninja  # [win]
+    - CMAKE_GENERATOR_TOOLSET=  # [win]
+    - CMAKE_GENERATOR_PLATFORM=  # [win]
+    - CMAKE_Fortran_COMPILER=%CONDA_PREFIX%\Library\bin\flang.exe  # [win]
```
Comment (isuruf) [inline recipes/pypartmc/meta.yaml:19]: Same here

File: recipes/pypartmc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  summary: Python interface to PartMC
+  dev_url: https://github.com/open-atmos/PyPartMC/
+  doc_url: https://open-atmos.github.io/PyPartMC/
+  license: GPL-2.0-or-later
+  license_file: LICENSE
+
+extra:
+  recipe-maintainers:
+    - a-v
+    - slayoo
```
Comment (carterbox) [inline recipes/pypartmc/meta.yaml:53]: Didn't complete the checklist! @slayoo has not consented to be a maintainer.

**Review by carterbox — CHANGES_REQUESTED** [review]
This package re-vendors and statically links to [all of these packages](https://github.com/open-atmos/PyPartMC/tree/main/gitmodules). This is not allowed (and mentioned in the checklist).

Each of these packages needs to be devendored/dynamically linked or a strong explanation should be made for each and then their license needs to be included in this package.

File: recipes/pypartmc/meta.yaml (hunk 3 above)
Comment (slayoo) [reply recipes/pypartmc/meta.yaml:53]: I do confirm.

**Comment by slayoo** [comment]
> This package re-vendors and statically links to [all of these packages](https://github.com/open-atmos/PyPartMC/tree/main/gitmodules). This is not allowed (and mentioned in the checklist).
> 
> Each of these packages needs to be devendored/dynamically linked or a strong explanation should be made for each and then their license needs to be included in this package.

Thank you for reviewing the PR!

Among the git submodules, we have several C++ header-only libraries, which are kind of by-design statically linked:
- pybind11 (https://github.com/pybind/pybind11/blob/master/LICENSE)
- pybind11_json (BSD 3-Clause License, https://github.com/pybind/pybind11_json/blob/master/LICENSE)
- span (Boost Software License 1.0, https://github.com/tcbrindle/span/blob/master/LICENSE_1_0.txt)
- string_view-standalone (MIT License, https://github.com/bitwizeshift/string_view-standalone/blob/master/LICENSE)

Then, we have several Fortran libraries for which package-distribution is hampered by compiler-dependent ABI:
- camp (MIT License, https://github.com/open-atmos/camp/blob/main/COPYING)
- json-fortran (https://github.com/jacobwilliams/json-fortran/blob/master/LICENSE)
- partmc (GNU General Public License v2.0, https://github.com/compdyn/partmc/blob/master/COPYING)

Finally, we have two libraries which we compile using our own CMakeLists file just using the source code from original archives to make it feasible to have it on Windows and macOS for which packages for these libraries (in tandem) are not readily available:
- SuiteSparse (https://github.com/DrTimothyAldenDavis/SuiteSparse/blob/master/LICENSE.txt) 
- sundials (BSD 3-Clause License, https://github.com/LLNL/sundials/blob/main/LICENSE)

**Comment by carterbox** [comment]
> Among the git submodules, we have several C++ header-only libraries, which are kind of by-design statically linked

Header only libraries can remain statically linked, but their licenses must be included in the license section of this recipe

> Then, we have several Fortran libraries for which package-distribution is hampered by compiler-dependent ABI

conda-forge packages all use the same compiler toolchain, this was one of the reasons for its creation.

> Finally, we have two libraries which we compile using our own CMakeLists file just using the source code from original archives to make it feasible to have it on Windows and macOS for which packages for these libraries (in tandem) are not readily available:

Suitesparse and sundials are both available for unix on conda-forge already. This recipe isn't being built for Windows either, so availability shouldn't be a problem.

**Comment by slayoo** [comment]
@a-v, I suggest we close this issue for now. The SUNDIALS and SuiteSparse static linking logic in our CMake files is not compliant with conda policy. Thank you all for your time and input!

**Pull request closed without merging**